### PR TITLE
feat(blueimp-load-image): Add promise style usage

### PR DESCRIPTION
**Pull request by zbarbuto**
As per [the latest docs](https://www.npmjs.com/package/blueimp-load-image#promise)

>If the loadImage() function is called without a callback function as second argument and the Promise API is available, it returns a Promise object

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://www.npmjs.com/package/blueimp-load-image#promise
- [x] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [x] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## blueimp-load-image ([<kbd>unpkg</kbd>](https://unpkg.com/browse/blueimp-load-image@latest/))
was missing the following properties:
1. requiresMetaData
1. fetchBlob
1. transform
1. global
1. readFile

<details>
<summary>as well as these 3 other properties...</summary>
<p>isInstanceOf, createObjectURL, revokeObjectURL</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 845d808d9040a203c428462c15f2787f63b7b19a
</p>

**Comment by pkonieczniak** [comment]
What's the state of this MR?

**Comment by zbarbuto** [comment]
The state is I made it a month ago and never heard anything back. I think the original maintainers are no longer active.

**Comment by zbarbuto** [comment]
Ping @ebk46 @KonstantinLukaschenko @moeinio

File: types/blueimp-load-image/index.d.ts [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
 // Project: https://github.com/blueimp/JavaScript-Load-Image
 // Definitions by: Evan Kesten <https://github.com/ebk46>
 //                 Konstantin Lukaschenko <https://github.com/KonstantinLukaschenko>
 //                 Saeid Rezaei <https://github.com/moeinio>
+//                 Zak Barbuto <https://github.com/zbarbuto>
 // Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
 
 declare namespace loadImage {
     type LoadImageCallback = (eventOrImage: Event | HTMLCanvasElement | HTMLImageElement, data?: MetaData) => void;
+    type LoadImageResult = MetaData & {
```
Comment (amcasey) [inline types/blueimp-load-image/index.d.ts:11]: I'm not familiar with this particular API, so please bear with me.  The pattern I usually see is that the callback variant accepts a callback with a `T` parameter and the promise variant returns a `Promise<T>`.  In this case, the callback and promise types look unrelated?  And the callback variant returns a value (rather than being void)?

**Review by amcasey — CHANGES_REQUESTED** [review]
Can you please help me understand the change?

File: types/blueimp-load-image/index.d.ts (hunk 1 above)
Comment (zbarbuto) [reply types/blueimp-load-image/index.d.ts:11]: I agree that `callback: (result: T) => void` and `Promise<T>` is typically what you see - but in the case of this library (and I've only got the docs to go by here) it seems to be a bit of a quirk that the callback takes two arguments (image and the result metadata) and the promise returns the result as a single payload:

image

image

image

Comment (amcasey) [reply types/blueimp-load-image/index.d.ts:11]: That's weird.  Thanks for doing the due diligence!  It's still not clear that callback version should have a non-void return type, but I'm willing to say that's out of scope for your change.

File: types/blueimp-load-image/blueimp-load-image-tests.ts [hunk 2]
```diff
@@ -22,11 +22,22 @@ const b64DataJPEG =
...
   canvas.toBlob((blob: Blob | null): void => {
     const url = canvas.toDataURL("image/png");
     console.log(url);
   });
-}, {canvas: true, orientation: true, maxWidth: 100, maxHeight: 100, crop: true});
+}, { canvas: true, orientation: true, maxWidth: 100, maxHeight: 100, crop: true });
+
+// Promise style
+loadImage(imageUrlJPEG, { canvas: true, orientation: true, maxWidth: 100, maxHeight: 100, crop: true }).then((data) => {
+  const canvas = (<any> data.image) as HTMLCanvasElement;
```
Comment (amcasey) [inline types/blueimp-load-image/blueimp-load-image-tests.ts:37]: Why is the cast to `any` needed?

Comment (zbarbuto) [reply types/blueimp-load-image/blueimp-load-image-tests.ts:37]: The existing typings have `false` as a possibility for `image` (if we don't use canvas I'm guessing)

image

Which means you can't directly type-cast a potential `boolean` into a `HTMLCanvasElement` without first casting to `unknown` or `any`

image

**Review by amcasey — APPROVED** [review]
The types seem weird to me, but you've clearly done your homework.

**Comment by zbarbuto** [comment]
Ready to merge
